kratos/sources/kernel.cpp

```diff
@@ -36,6 +38,9 @@ namespace Kratos
     std::cout << "           Multi-Physics "<< KRATOS_VERSION << std::endl;
 
         mKratosApplication.RegisterVariables();
+        
+        RegisterLinearSolvers();
+        RegisterPreconditioners();
```

Comment:
I would strongly avoid calling these functions in the constructor of the Kernel. They are not exception free and with a lot of background details.....

Reply:
I see your point. I think it was needed to put it there, but i don't remember the details. i'll check when i have some time